### PR TITLE
chore: Refactor Field into a standalone component

### PR DESCRIPTION
**Pull request by behowell**
## Related Issue(s)

* Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                    Change |
| :--------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/CheckboxField.fixture.js'>CheckboxField</abbr>       |  `39.26 kB`<br />`11.964 kB` |   `39.1 kB`<br />`11.901 kB` | `-160 B` <br />`-63 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/ComboboxField.fixture.js'>ComboboxField</abbr>       | `78.578 kB`<br />`25.979 kB` | `78.477 kB`<br />`25.933 kB` | `-101 B` <br />`-46 B`  |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/InputField.fixture.js'>InputField</abbr>                |  `34.41 kB`<br />`10.891 kB` | `34.309 kB`<br />`10.863 kB` | `-101 B` <br />`-28 B`  |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressField.fixture.js'>ProgressField</abbr>       |  `23.404 kB`<br />`8.201 kB` |  `23.207 kB`<br />`8.143 kB` | `-197 B` <br />`-58 B`  |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroupField.fixture.js'>RadioGroupField</abbr>      |  `24.535 kB`<br />`8.931 kB` |  `24.398 kB`<br />`8.917 kB` | `-137 B` <br />`-14 B`  |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/SelectField.fixture.js'>SelectField</abbr>             | `32.737 kB`<br />`10.583 kB` | `32.636 kB`<br />`10.552 kB` | `-101 B` <br />`-31 B`  |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/SliderField.fixture.js'>SliderField</abbr>             | `42.202 kB`<br />`13.241 kB` | `42.099 kB`<br />`13.208 kB` | `-103 B` <br />`-33 B`  |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButtonField.fixture.js'>SpinButtonField</abbr> | `53.786 kB`<br />`15.401 kB` |  `53.69 kB`<br />`15.347 kB` |  `-96 B` <br />`-54 B`  |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/SwitchField.fixture.js'>SwitchField</abbr>             |  `34.286 kB`<br />`10.41 kB` | `34.192 kB`<br />`10.382 kB` |  `-94 B` <br />`-28 B`  |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/TextareaField.fixture.js'>TextareaField</abbr>       | `35.893 kB`<br />`11.307 kB` | `35.791 kB`<br />`11.273 kB` | `-102 B` <br />`-34 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/Checkbox.fixture.js'>Checkbox</abbr>                                                                              |  `32.658 kB`<br />`10.169 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr>                                                 |  `80.888 kB`<br />`25.842 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                                                 |  `79.674 kB`<br />`25.637 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `59.888 kB`<br />`16.652 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `187.539 kB`<br />`52.638 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `34.379 kB`<br />`11.322 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                                                                       |   `24.673 kB`<br />`8.065 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.069 kB`<br />`2.053 kB` |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressBar.fixture.js'>ProgressBar</abbr>                                                                        |   `13.578 kB`<br />`5.235 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                                                                                       |  `32.247 kB`<br />`10.468 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                                                                             |   `14.718 kB`<br />`5.886 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                                                                    |   `23.963 kB`<br />`8.206 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                                                                    |  `32.663 kB`<br />`10.403 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                                                                        |  `44.989 kB`<br />`12.975 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                                                                    |   `27.791 kB`<br />`8.605 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                                                                                 | `129.317 kB`<br />`35.207 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `42.971 kB`<br />`11.781 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>                                                                 | `127.563 kB`<br />`32.864 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr>                                                         |      `81 kB`<br />`20.124 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>                                                                   |   `80.429 kB`<br />`19.97 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr>                                                                              |   `26.156 kB`<br />`8.508 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d2818f204f2bd95093960b3ce69e926556ffb6df'>d2818f204f2bd95093960b3ce69e926556ffb6df</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/Avatar-mount_base.html">1308</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/Avatar-mount.html">1297</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/Button-mount_base.html">943</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/Button-mount.html">932</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/FluentProvider-mount_base.html">1606</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/FluentProvider-mount.html">1576</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/FluentProviderWithTheme-mount_base.html">619</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/FluentProviderWithTheme-mount.html">639</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">596</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">582</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">625</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">624</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/InfoButton-mount_base.html">567</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/InfoButton-mount.html">553</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/MakeStyles-mount_base.html">1904</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/MakeStyles-mount.html">1899</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/Persona-mount_base.html">2951</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/Persona-mount.html">2957</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/SpinButton-mount_base.html">2462</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26416/perf-test-react-components/SpinButton-mount.html">2487</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-checkbox/etc/react-checkbox.api.md [hunk 1]
```diff
@@ -8,31 +8,25 @@
```
Comment (fabricteam) [inline packages/react-components/react-checkbox/etc/react-checkbox.api.md:8]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 19 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=26416&commitId=dc25f851486a41e340994e05c61eabb7243174ee&env=prod)

✅ There was 0 screenshots added, 0 screenshots removed, 1756 screenshots unchanged, 0 screenshots with different dimensions and 19 screenshots with visible difference.


  <details><summary><b>unknown</b> 19 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|ComboboxField converged.required.default.chromium.png|6| Changed|
|ComboboxField converged.size-large.default.chromium.png|1866| Changed|
|ComboboxField converged.size-small.default.chromium.png|2698| Changed|
|InputField converged.required.default.chromium.png|6| Changed|
|InputField converged.size-large.default.chromium.png|1791| Changed|
|InputField converged.size-small.default.chromium.png|2554| Changed|
|ProgressField converged.horizontal.default.chromium.png|12| Changed|
|ProgressField converged.required.default.chromium.png|6| Changed|
|ProgressField converged.validation.default.chromium.png|57| Changed|
|RadioGroupField converged.required.default.chromium.png|6| Changed|
|SelectField converged.required.default.chromium.png|6| Changed|
|SelectField converged.size-large.default.chromium.png|1899| Changed|
|SelectField converged.size-small.default.chromium.png|2639| Changed|
|SliderField converged.required.default.chromium.png|6| Changed|
|SpinButtonField converged.required.default.chromium.png|6| Changed|
|SwitchField converged.required.default.chromium.png|6| Changed|
|TextareaField converged.required.default.chromium.png|6| Changed|
|TextareaField converged.size-large.default.chromium.png|1785| Changed|
|TextareaField converged.size-small.default.chromium.png|2565| Changed|

</details>


<div id="vrtCommentfluentuiv9"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: apps/vr-tests-react-components/src/stories/Field.stories.tsx [hunk 2]
```diff
@@ -14,9 +14,11 @@ import { SpinButtonField_unstable as SpinButtonField } from '@fluentui/react-spi
 import { SwitchField_unstable as SwitchField } from '@fluentui/react-switch';
```
Comment (fabricteam) [inline apps/vr-tests-react-components/src/stories/Field.stories.tsx:14]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 60 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=26416&commitId=7f492c9ca60ef7358598421bc4deb8a978d3b6be&env=prod)

✅ There was 0 screenshots added, 0 screenshots removed, 1715 screenshots unchanged, 0 screenshots with different dimensions and 60 screenshots with visible difference.


  <details><summary><b>unknown</b> 60 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|CheckboxField converged.base.default.chromium.png|256347| Changed|
|CheckboxField converged.fieldLabel.default.chromium.png|261506| Changed|
|CheckboxField converged.hint.default.chromium.png|260018| Changed|
|CheckboxField converged.horizontal.default.chromium.png|273110| Changed|
|CheckboxField converged.required.default.chromium.png|256341| Changed|
|CheckboxField converged.size-large.default.chromium.png|257201| Changed|
|CheckboxField converged.validation.default.chromium.png|329471| Changed|
|ComboboxField converged.base.default.chromium.png|261522| Changed|
|ComboboxField converged.hint.default.chromium.png|267511| Changed|
|ComboboxField converged.horizontal.default.chromium.png|270723| Changed|
|ComboboxField converged.required.default.chromium.png|261516| Changed|
|ComboboxField converged.size-large.default.chromium.png|264599| Changed|
|ComboboxField converged.size-small.default.chromium.png|257956| Changed|
|ComboboxField converged.validation.default.chromium.png|366437| Changed|
|InputField converged.base.default.chromium.png|261569| Changed|
|InputField converged.hint.default.chromium.png|267558| Changed|
|InputField converged.horizontal.default.chromium.png|270792| Changed|
|InputField converged.required.default.chromium.png|261562| Changed|
|InputField converged.size-large.default.chromium.png|264630| Changed|
|InputField converged.size-small.default.chromium.png|258014| Changed|
|InputField converged.validation.default.chromium.png|366630| Changed|
|ProgressField converged.base.default.chromium.png|253836| Changed|
|ProgressField converged.hint.default.chromium.png|257572| Changed|
|ProgressField converged.horizontal.default.chromium.png|270800| Changed|
|ProgressField converged.required.default.chromium.png|253830| Changed|
|ProgressField converged.validation.default.chromium.png|318394| Changed|
|RadioGroupField converged.base.default.chromium.png|286206| Changed|
|RadioGroupField converged.hint.default.chromium.png|293316| Changed|
|RadioGroupField converged.horizontal.default.chromium.png|290467| Changed|
|RadioGroupField converged.required.default.chromium.png|286200| Changed|
|RadioGroupField converged.validation.default.chromium.png|456699| Changed|
|SelectField converged.base.default.chromium.png|261455| Changed|
|SelectField converged.hint.default.chromium.png|267444| Changed|
|SelectField converged.horizontal.default.chromium.png|270699| Changed|
|SelectField converged.required.default.chromium.png|261449| Changed|
|SelectField converged.size-large.default.chromium.png|264536| Changed|
|SelectField converged.size-small.default.chromium.png|257912| Changed|
|SelectField converged.validation.default.chromium.png|366195| Changed|
|SliderField converged.base.default.chromium.png|261410| Changed|
|SliderField converged.hint.default.chromium.png|267399| Changed|
|SliderField converged.horizontal.default.chromium.png|270740| Changed|
|SliderField converged.required.default.chromium.png|261404| Changed|
|SliderField converged.validation.default.chromium.png|366110| Changed|
|SpinButtonField converged.base.default.chromium.png|261526| Changed|
|SpinButtonField converged.hint.default.chromium.png|267515| Changed|
|SpinButtonField converged.horizontal.default.chromium.png|270741| Changed|
|SpinButtonField converged.required.default.chromium.png|261519| Changed|
|SpinButtonField converged.validation.default.chromium.png|366453| Changed|
|SwitchField converged.base.default.chromium.png|262689| Changed|
|SwitchField converged.hint.default.chromium.png|268981| Changed|
|SwitchField converged.horizontal.default.chromium.png|270976| Changed|
|SwitchField converged.required.default.chromium.png|262683| Changed|
|SwitchField converged.validation.default.chromium.png|372451| Changed|
|TextareaField converged.base.default.chromium.png|270008| Changed|
|TextareaField converged.hint.default.chromium.png|277852| Changed|
|TextareaField converged.horizontal.default.chromium.png|274583| Changed|
|TextareaField converged.required.default.chromium.png|270002| Changed|
|TextareaField converged.size-large.default.chromium.png|276484| Changed|
|TextareaField converged.size-small.default.chromium.png|264207| Changed|
|TextareaField converged.validation.default.chromium.png|400692| Changed|

</details>


<div id="vrtCommentfluentuiv9"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

**Comment by behowell** [comment]
Closing in favor of #26430

**Pull request closed without merging**